stripe/src/main/java/com/stripe/android/paymentsheet/model/ViewState.kt

```diff
@@ -1,16 +1,36 @@
 package com.stripe.android.paymentsheet.model
 
 import com.stripe.android.PaymentIntentResult
+import com.stripe.android.paymentsheet.PaymentOptionResult
 
-sealed class ViewState {
-    data class Ready(
-        val amount: Long,
-        val currencyCode: String
-    ) : ViewState()
+internal sealed class ViewState {
 
-    object Confirming : ViewState()
+    internal sealed class PaymentSheet : ViewState() {
+        data class Ready(
+            val amount: Long,
+            val currencyCode: String
+        ) : PaymentSheet()
 
-    data class Completed(
-        val paymentIntentResult: PaymentIntentResult
-    ) : ViewState()
+        object Confirming : PaymentSheet()
+
+        data class Completed(
+            val result: PaymentIntentResult
+        ) : PaymentSheet()
+    }
+
+    internal sealed class PaymentOptions : ViewState() {
+        object Ready : PaymentOptions()
+
+        object Confirming : PaymentOptions()
+
+        // Completed indicates the animation should show it completed
+        data class Completed(
+            val result: PaymentOptionResult
+        ) : PaymentOptions()
```

Comment:
With `Finished`, would it be helpful to rename these to `StartProcessing` and `EndProcessing`?

Reply:
yeah, I was thinking the same thing!  I forgot, and I am now glad you mentioned it.